### ISSUE TITLE
Parse errors can result in a NPE in the SuppressWarningsHolder

### ISSUE THREAD

**Issue by jshiell** [checkstyle/checkstyle#1489]
If you have a rules file with a suppression filter configured and you attempt to parse a syntactically invalid file then a NPE will be thrown from the `SuppressWarningsHolder.isSuppressed` method as the `ENTRIES` ThreadLocal has never been populated.

This occurs as the parse exception in `TreeWalker.processFiltered` means that the `walk` methods are never called, and hence the `SuppressWarningsHolder.beginTree` method is never invoked. Then when `Checker.process` invokes `fireErrors` we hit the `isSuppressed` method and it all goes tits up.
```
Caused by: java.lang.NullPointerException
    at com.puppycrawl.tools.checkstyle.checks.SuppressWarningsHolder.isSuppressed(SuppressWarningsHolder.java:200)
    at com.puppycrawl.tools.checkstyle.filters.SuppressWarningsFilter.accept(SuppressWarningsFilter.java:38)
    at com.puppycrawl.tools.checkstyle.api.FilterSet.accept(FilterSet.java:88)
    at com.puppycrawl.tools.checkstyle.Checker.fireErrors(Checker.java:358)
    at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:276)
```
This isn't visible to users in the command line tool as the tool calls `System.exit` in the finally block, thus suppressing the exception. And as the parse exception has already been logged the user sees a sane output. However, it's a bit painful for other users of the `Checker.process` method, such as plugins (this [CheckStyle-IDEA](https://github.com/jshiell/checkstyle-idea/issues/127) issue led me to this problem).

I'm able to reproduce this using any rules file with a `SuppressionFilter` and a syntactically incorrect file, such as this example contributed by @twbecker:
```
public class Test {
    public static void main(String[] args) {
        private long foo;
        protected int bar = 1;
        private int
    }

    public static class Inner {
        private Long baz;
    }
}
```

Thanks :smile:

**Comment by jsotuyod** [comment]
I have just came accross this very exception, in a completely different environment.

I injected Checkstyle 6.9into the latest maven checkstyle plugin on Maven 3. Running a simple `mvn checkstyle:checkstyle` works like a charm, BUT the moment I tell Maven to run modules in parallel: havoc.

`mvn checkstyle:checkstyle -T1C`

> Caused by: java.lang.NullPointerException
>	at com.puppycrawl.tools.checkstyle.checks.SuppressWarningsHolder.isSuppressed(SuppressWarningsHolder.java:207)
>	at com.puppycrawl.tools.checkstyle.filters.SuppressWarningsFilter.accept(SuppressWarningsFilter.java:38)
	at com.puppycrawl.tools.checkstyle.api.FilterSet.accept(FilterSet.java:88)
	at com.puppycrawl.tools.checkstyle.Checker.fireErrors(Checker.java:353)
	at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:271)
	at org.apache.maven.plugin.checkstyle.exec.DefaultCheckstyleExecutor.executeCheckstyle(DefaultCheckstyleExecutor.java:253)
	at org.apache.maven.plugin.checkstyle.AbstractCheckstyleReport.executeReport(AbstractCheckstyleReport.java:473)
	at org.apache.maven.plugin.checkstyle.CheckstyleReport.executeReport(CheckstyleReport.java:155)
	at org.apache.maven.reporting.AbstractMavenReport.generate(AbstractMavenReport.java:255)
	at org.apache.maven.reporting.AbstractMavenReport.execute(AbstractMavenReport.java:143)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:101)
	... 12 more

**Comment by ndbernardi** [comment]
+1, seeing the same on Maven, but in my case not even disabling parallel execution fixes the issue...

**Comment by romani** [comment]
checkstyle work only on compilable sources.

more details at https://github.com/checkstyle/checkstyle/pull/2036#issuecomment-136481616

**Comment by jsotuyod** [comment]
My issue is on source that is compiling.

Doing `mvn clean package checkstyle:checkstyle -T1C` will break at checkstyle, not at compile stage.

**Comment by romani** [comment]
@jsotuyod , please create another issue and provide source file and config, and maybe even a whole project, but most likely problem should be reported to maven plugin first.
